src/php/Compiler/Lexer/Lexer.php

```diff
@@ -36,10 +36,12 @@ final class Lexer implements LexerInterface
     private int $line = 1;
     private int $column = 0;
     private string $combinedRegex;
+    private bool $withoutLocation;
 
-    public function __construct()
+    public function __construct(bool $withoutLocation = false)
```

Comment:
What about using `bool $withLocation = true`?

Reply:
Will change this.